### DISCUSSION TITLE
Cron task not running at all

### DISCUSSION THREAD

**Discussion by iandrianto** [directus/directus#16004] [Archive]
I'm using Directus 9.18.1
I want to add a hook that runs periodically with cron syntax.

I create a file in _hooks/hook1/index.js_ whose content is similar to [this discussion](https://github.com/directus/directus/discussions/4120).
```
module.exports = () => {
  return {
    'cron(* * * * *)': async () => {
      console.log('running a task every minute');
    },
  };
};
```

But it's not running at all. Anything wrong?

And please fix the overall documentation and examples on https://docs.directus.io/ because many of them are very unclear (for example, not giving example where to put the file and the code of many examples cannot run)

**Comment by br41nslug** [comment]
It looks like you're using an old format of building extensions which is not supported since Directus is out of the Release Candidate phase (the breaking change was made here https://github.com/directus/directus/releases/tag/v9.0.0-rc.101 in this pr https://github.com/directus/directus/pull/8027).
In the new format this extension would look like this (in commonjs ignoring the extension-sdk for now):
```js
module.exports = ({ schedule }, { logger }) => {
  schedule('* * * * *', async () => {
      logger.info('running a task every minute');
   });
};
```

> and the code of many examples cannot run

Did you go through the introduction to extensions? https://docs.directus.io/extensions/creating-extensions.html
The JS format above is CommonJS which works because it is natively supported by NodeJS but not actively supported by Directus. To allow for more modern ESM development we provide an SDK with build tooling all examples work with the SDK and so need to be compiled before they run in node.
